### PR TITLE
kupfer: init at 319

### PR DESCRIPTION
**Pull request by cobbal**
###### Motivation for this change

Added a package for [kupfer](https://kupferlauncher.github.io/).

###### Things done

- [ ] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] Linux
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

I think that the closure is a little bloated, because I'm grabbing the $PYTHONPATH from the build environment instead of using the wrapPython scripts, but I couldn't figure out how to get them to work when the binaries are wrappers themselves.

### CODE REVIEW

File: pkgs/applications/misc/kupfer/default.nix [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    wrapProgram $out/bin/kupfer \
+        --prefix GI_TYPELIB_PATH : "$GI_TYPELIB_PATH" \
+        --prefix PYTHONPATH : "$PYTHONPATH" \
+        --set PYTHONNOUSERSITE 1
+  '';
+
+  meta = with lib; {
+    description = "A smart, quick launcher";
+    homepage    = "https://kupferlauncher.github.io/";
+    license     = licenses.gpl3;
```
Comment (Mic92) [inline pkgs/applications/misc/kupfer/default.nix]: Do you want to maintain this package?

File: pkgs/applications/misc/kupfer/default.nix [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    runHook postInstall
+  '';
+
+  doCheck = false; # no tests
+
+  meta = with stdenv.lib; {
+    description = "A smart, quick launcher";
+    homepage    = "https://kupferlauncher.github.io/";
+    license     = licenses.gpl3;
+    platforms   = platforms.linux;
```
Comment (Mic92) [inline pkgs/applications/misc/kupfer/default.nix]: Do you want to maintain this package?

**Comment by cobbal** [comment]
@Mic92 Thanks for the fixes. I've patched up the runtime dependencies and tested that it still runs. I also added myself as the maintainer.

**Comment by Mic92** [comment]
Thanks!

**Pull request merged**